src/model/entities/rundown.ts

```diff
@@ -150,8 +153,15 @@ export class Rundown extends BasicRundown {
 
   private setNextFromActive(owner: Owner): void {
     this.unmarkNextPart()
-
     if (!this.activeCursor) {
+      try {
+        this.unmarkNextSegment()
+        this.setFirstSegmentAndPartNextCursor()
+      } catch (exception) {
+        if (!(exception instanceof NotFoundException)) {
+          throw exception
+        }
+      }
```

Comment:
This seems like some good logic to have tests for ;) 
I'm available if you want a hand in writing the tests.

Reply:
Well i'll be damned if you are not correct sir. I added tests for removing the last segment and thus setting the next cursor to undefined since we have nothing left.
There is also now a test for when we add back a single segment to an empty active rundown, in which case it should be the one set as next.